Client/Frontend/Browser/BrowserViewController.swift

```diff
@@ -1759,6 +1759,53 @@ class BrowserViewController: UIViewController {
         duckDuckGoPopup = popup
         popup.showWithType(showType: .flyUp)
     }
+    
+    private func showBookmarkController() {
+        let bookmarkViewController = BookmarksViewController(
+            folder: nil,
+            isPrivateBrowsing: PrivateBrowsingManager.shared.isPrivateBrowsing)
+        
+        bookmarkViewController.toolbarUrlActionsDelegate = self
+        
+        presentSettingsNavigation(with: bookmarkViewController)
+    }
+    
+    func openAddBookmark() {
+        guard let selectedTab = tabManager.selectedTab,
+              let selectedUrl = selectedTab.url,
+              !(selectedUrl.isLocal || selectedUrl.isReaderModeURL) else {
+            return
+        }
+        
+        let bookmarkUrl = selectedUrl.decodeReaderModeURL ?? selectedUrl
+        
+        let mode = BookmarkEditMode.addBookmark(title: selectedTab.displayTitle, url: bookmarkUrl.absoluteString)
+        
+        let addBookMarkController = AddEditBookmarkTableViewController(mode: mode)
+        
+        presentSettingsNavigation(with: addBookMarkController, cancelEnabled: true)
+    }
+    
+    private func presentSettingsNavigation(with controller: UIViewController, cancelEnabled: Bool = false) {
+        let navigationController = SettingsNavigationController(rootViewController: controller)
+        navigationController.modalPresentationStyle = .formSheet
+        
+        let cancelBarbutton = UIBarButtonItem(
+            barButtonSystemItem: .cancel,
+            target: navigationController,
+            action: #selector(SettingsNavigationController.done))
+        
+        let doneBarbutton = UIBarButtonItem(
```

Comment:
is this button needed?
In `AddEditBookmarksVC` we place save button instead

Reply:
yeah we need it cause we joined the functions there that does the presenting, other we will not have done button for the main bookmark screen

![1](https://user-images.githubusercontent.com/6643505/98822264-9f86a980-23fe-11eb-8239-cd8cb8cf68bf.png)